components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1131,11 +1148,37 @@ private class MonitoredItemWrapper {
                             EventTemplate.PendingAlarms.RetainIndex = eventFilter.SelectClauses.Count();
                             eventFilter.AddSelectClause(ObjectTypeIds.ConditionType, "Retain");
                         }
+
+                        // set up the timers
+                        if (EventTemplate.PendingAlarms.UpdateIntervalTimespan.HasValue) {
+                            _pendingAlarmsUpdateTimer.Interval = EventTemplate.PendingAlarms.UpdateIntervalTimespan.Value.TotalMilliseconds;
+                            _pendingAlarmsUpdateTimer.Elapsed += PendingAlarmsUpdateTimer_Elapsed;
+                            _pendingAlarmsUpdateTimer.Start();
+                        }
+                        if (EventTemplate.PendingAlarms.SnapshotIntervalTimespan.HasValue) {
+                            _pendingAlarmsSnapshotTimer.Interval = EventTemplate.PendingAlarms.SnapshotIntervalTimespan.Value.TotalMilliseconds;
+                            _pendingAlarmsSnapshotTimer.Elapsed += PendingAlarmsSnapshotTimer_Elapsed;
+                            _pendingAlarmsSnapshotTimer.Start();
+                        }
                     }
                     Item.Filter = eventFilter;
                 }
             }
 
+            private void PendingAlarmsUpdateTimer_Elapsed(object sender, System.Timers.ElapsedEventArgs e) {
+                if (EventTemplate?.PendingAlarms?.Dirty == true) {
+                    _pendingAlarmsUpdateTimer.Enabled = false;
```

Comment:
Perhaps we should validate this behavior. I see what you are doing but let's discuss if this is the intended behavior of the timers from the use case.

Reply:
Yes, intentionally I created this draft PR so we can share and discuss. I'm going to ask the customer as well.

Reply:
UPDATE: Talked to the customer and we agreed on the business logic. Revisit the PR for additional comments